### ISSUE TITLE
way to not only add classes but elements

### ISSUE THREAD

**Issue by logemann** [chrisg86/gatsby-remark-classes#6]
super nice plugin. Thanks for providing it. But is there a way to do something like this:

md:
```
* foo
* bar
```

turns into

html:
```
<ul class="myclass">
<li><i class="fa fa-check"/>foo</li>
<li><i class="fa fa-check"/>bar</li>
</ul>
```

The ul class part is of course easy, the plugin is developed exactly for that. But i wonder if one could add a complete new element to an element. Here the <i> which represents an icon from FontAwesome.

Thanks

**Comment by chrisg86** [comment]
Thank you for the interesting idea! Directly injecting markup however, would be out of scope for this plugin. 

You can still make this work with css pseudo-selectors, such as `:before` or `:after`. I've quickly set up a fresh gatsby site, configured the plugin and was able to make it work with said pseudo selectors. 

TLDR; Here are the details:

Created a `templates` directory with `blogTemplate.js` as outlined in the official gatsby docs:
https://www.gatsbyjs.com/docs/how-to/routing/adding-markdown-pages/#create-a-page-template-for-the-markdown-files

In `blogTemplate.js` add the import for `'react-helmet'` so you can load in Font-Awesome:
```js
import React from "react"
import { graphql } from "gatsby"
import { Helmet } from "react-helmet"

export default function Template({
  data, // this prop will be injected by the GraphQL query below.
}) {
  const { markdownRemark } = data // data.markdownRemark holds your post data
  const { frontmatter, html } = markdownRemark
  return (
    <div className="blog-post-container">
      <Helmet>
        <link
          href="https://unpkg.com/tailwindcss@^2/dist/tailwind.min.css"
          rel="stylesheet"
        />
        <link
          href="https://unpkg.com/font-awesome@4.7.0/css/font-awesome.min.css"
          rel="stylesheet"
        />
      </Helmet>
      <div className="blog-post">
        <div
          className="blog-post-content"
          dangerouslySetInnerHTML={{ __html: html }}
        />
      </div>
    </div>
  )
}
export const pageQuery = graphql`
  query($slug: String!) {
    markdownRemark(frontmatter: { slug: { eq: $slug } }) {
      html
      frontmatter {
        date(formatString: "MMMM DD, YYYY")
        slug
        title
      }
    }
  }
`
```

Inside my `blogTemplate.js`, I've also added an import to a css file as so:
```js
import "./test.css"
```

Create that file and paste the following content:
```css
.myCustomList {
  padding-left: 1rem;
  margin-bottom: 0.125rem;
}

.myCustomList::before {
  content: "\f061";         /* Code for right pointing arrow */
  display: inline-block;
  font-family: FontAwesome;
  margin-left: -0.75rem;    /* Determines the offset on the left side */
  width: 1.25rem;           /* Determines the distance between icon and text */
}

```

In your `gatsby-config.js`, configure `gatsby-remark-classes`:
```js
{
  resolve: `gatsby-transformer-remark`,
  options: {
    plugins: [
      {
        resolve: `gatsby-remark-classes`,
        options: {
          classMap: {
            "list[ordered=false]": "list-none p-0",
            listItem: "myCustomList", // Make sure this matches the class name in the css file
          },
        },
      },
    ],
  },
};
```

Try it out with a list in your markdown file:
```md
---
slug: "/blog/my-first-post"
date: "2021-01-20"
title: "My first blog post"
---

- Hi
- Oh, hello!
```

That should be all, you should see something like this:
<img width="180" alt="custom-list" src="https://user-images.githubusercontent.com/4715597/105176460-0cf03e00-5b04-11eb-81ed-6d2889a062a0.png">

**Comment by logemann** [comment]
Ohhh. this is awesome. I totally forgot pseudo selectors being a solution here.... thanks.

**Comment by chrisg86** [comment]
Happy I could help!
